### PR TITLE
[WIP] Fix "Sortable table extends awkwardly" bug

### PR DESCRIPTION
**Pull request by efalayi**
Resolves [#2878](https://github.com/reactioncommerce/reaction/issues/2878)
Table column expands outside of the panel.

Some changes were made in -https://github.com/reactioncommerce/reaction/commit/0c465809a27bae8b5dcbffefa60c2033b12e842a#diff-1462e7801c36243a6542abe7414fe30bL36
-https://github.com/reactioncommerce/reaction/commit/0c465809a27bae8b5dcbffefa60c2033b12e842a#diff-1462e7801c36243a6542abe7414fe30bL163. These are the default style of `sortableTable` component.

###### Fix
```
- Add overflow property to `rt-table` class
- Fix border styles
```

###### Test
```
Place some orders
Login in as admin
Go to orders
Grab and expand one of the column headers
See that column does not extend out of the table.
```

### CODE REVIEW

**Comment by efalayi** [comment]
@kieckhafer this is not a permanent fix. What do you think about the approach? I could create a function that handles setting min widths for the different columns.

**Comment by rymorgan** [comment]
Yea, it would great if this fix would work across all the react tables, I'm guessing this issue happens everywhere this table is used. 

pepke6cbyb

**Comment by efalayi** [comment]
@rymorgan at the moment, i'm implementing this fix for Orders table. We might have to create a separate issue to accommodate changes to other the instances of sortableTable.

**Comment by spencern** [comment]
@efalayi Can you explain why the fix only works for one table and not all sortableTable components? If this PR is hacky and there's a "right" way to do it, I'd much prefer we just do it right the first time.

**Comment by rymorgan** [comment]
Yea, I am hoping we are building it modular enough to work elsewhere.

**Comment by efalayi** [comment]
@spencern I observed that min and max widths were specified for columns in Orders table. This fix handled it. But we can just specify the min and max widths in `sortableTable` (the main component which renders react table in RC) directly. Should I do that in this PR?

**Comment by spencern** [comment]
> @kieckhafer this is not a permanent fix. What do you think about the approach? I could create a function that handles setting min widths for the different columns.

I don't think we should be doing temporary fixes. If you're not sure what to do, please reach out to someone for help.

**Comment by rymorgan** [comment]
The current change isn't fixing the issue. 

ljnujmcv4p

**Comment by efalayi** [comment]
@rymorgan just observed that happens after a certain width. I'll look into it.

**Comment by spencern** [comment]
@efalayi any updates on this? I'd like to get this merged today.

File: imports/plugins/core/orders/client/components/orderTable.js [hunk 1]
```diff
@@ -371,7 +372,7 @@ class OrderTable extends Component {
           />
         }
         <SortableTable
-          tableClassName={`rui order table -highlight ${this.props.selectedItems.length > 0 ?
+          tableClassName={`rui order table -striped -highlight ${this.props.selectedItems.length > 0 ?
```
Comment (mikemurray) [inline imports/plugins/core/orders/client/components/orderTable.js]: Whats the purpose of the `-`(dash) in front of the class names for `striped` and `highlight`. Seems unnecessary to me? @kieckhafer thoughts?

Comment (kieckhafer) [reply imports/plugins/core/orders/client/components/orderTable.js]: That's just how ReactTable does it, it's not on our end.

**Comment by efalayi** [comment]
@spencern changes I've made don't seem to override the styles set. I'm still working on it.

**Comment by joykare** [comment]
@efalayi adding https://github.com/reactioncommerce/reaction/commit/0c465809a27bae8b5dcbffefa60c2033b12e842a#diff-1462e7801c36243a6542abe7414fe30bL36
https://github.com/reactioncommerce/reaction/commit/0c465809a27bae8b5dcbffefa60c2033b12e842a#diff-1462e7801c36243a6542abe7414fe30bL163
those two lines back fixes the issue. I'm not sure whether it breaks anything though. @foladipo any input?

**Comment by foladipo** [comment]
@joykare  @efalayi I tested this and it currently works fine for me. That is, when I expand any of the columns, the overall table becomes horizontally scrollable to accomodate the extra width, which I think is acceptable behaviour.

The only problem I found is that the horizontal dividers in the table don't expand with it. See SS below for details:
screen shot 2017-10-12 at 11 58 22 am-highlighted

**Comment by rymorgan** [comment]
I'm looking at this PR and it's still not working the intended way. I think there is some confusion about how this should work.

I want the columns to never move outside the table, but rather condense down to a min width of around 20pixels.

So you can expand the columns to see hidden information w/out losing the rest of the columns.

Here is what I'm seeing now. 
qgmbvmukfz

**Comment by spencern** [comment]
@efalayi why was this PR closed?

**Comment by efalayi** [comment]
@spencern I made an error. I'll reopen it right away.

**Comment by spencern** [comment]
Hi @efalayi 
Closing a PR that isn't ready yet isn't a problem as long as you leave a note explaining why it was closed and what steps you'll be taking to get it ready to open again.

**Comment by zenweasel** [comment]
@efalayi Please update this with status and next steps.

**Comment by efalayi** [comment]
@zenweasel at the moment I've reverted to previous styles implemented on the table. Next step is find a way to improve `SortableTable` component to use different styles.

**Comment by efalayi** [comment]
@joykare I've reverted to the old styles on this table. We might have to implement a different style of `sortableTable` component for ordersTable. What do you think?

**Comment by rymorgan** [comment]
@kieckhafer @joykare It doesn't seem like we need a whole new table to make the changes I requested. Can y'all help @efalayi out or let me know that my request is harder than I think to implement.

**Comment by spencern** [comment]
To echo @rymorgan. Let's either get this resolved and merged or close it and let someone else work on it. This PR has been [WIP] for a long time.

**Comment by efalayi** [comment]
@spencern this issue will resolved before the end of the week. I got caught with another issue.  @joykare @kieckhafer could we have a sync on this issue?

**Comment by spencern** [comment]
@joykare I think you may be best suited to pair with @efalayi as @kieckhafer has a pretty long list of things to get done right now.

**Comment by efalayi** [comment]
I read the source code of `react-table` package and I discovered that:
1. `min-width` can be hard-coded.
1. `max-width` is dynamically set when a resize function is called.
1. passing `maxWidth` and `minWidth` as props doesn't override the previous css styles.

Looking at the table for the orders dashboard, I noticed that some columns do not really need to be resizable. Since the resize function can be disabled in `react-table`, what if we disabled the resizing of some (or all) columns and then display extra text with a tooltip? @joykare  @kieckhafer @spencern what do you think?

**Comment by kieckhafer** [comment]
What kind of extra text would you be displaying with a tooltip? How would that work?

I think a big issue here is screen-size, the columns are being hidden / funky on smaller screens. You can't use tooltips on mobile, at least not to be the only display of information.

**Comment by rymorgan** [comment]
@efalayi I don't think tooltip is a good solution here. It'll be a pain for people to use and won't work on mobile. @joykare was working on a solution the rows were horizontally scrollable. Which could be the best approach and would work on mobile. We'd have to fix the first column and last columns so that they aren't scrollable and the rest of the columns scroll underneath them in both directions. @joykare Should be able to help you as she had this partially working when she was in Los Angeles.

**Comment by joykare** [comment]
@rymorgan @efalayi I was working on this got caught up in the foray sprint and travelling. But I understand what @rymorgan wants so we can sync on this @efalayi.

**Comment by efalayi** [comment]
@rymorgan @kieckhafer you're right. Tooltip wouldn't work for small screens. @joykare when would you like us to have a sync on this?

**Comment by efalayi** [comment]
@joykare and I will have a sync on this today. Updates will be given later today.

**Comment by efalayi** [comment]
### Update
@joykare  and I concluded that the first and last columns would be fixed (not resizable) but every other column in between them would be resizable. This would be implemented for all instances of `sortableTable`.

**Comment by rymorgan** [comment]
@efalayi I'm not sure we want to implement this in every react table we have. We'd definitely need to look at that before rolling it out. I'd rather just apply this to this one table.

**Comment by efalayi** [comment]
@rymorgan @joykare I updated this PR some days ago. Could you check it out to see the changes I've made.

**Comment by rymorgan** [comment]
@efalayi Did you get together with @joykare on this issue? I'm still not happy with this solution. Especially on mobile. You can currently expand the table out and not be able pull the fields back after you expand them. This will amount to a frustrating UX. What happened the idea we discussed with @joykare with fixing the first and last columns and allowing the middle columns to be scrollable? 

A few things to note on this: 
1. We can lose columns completely once they are expanded. 
2. The button to process the orders should be fixed b/c the most important action a user will take on this page is to hit that button and process an order. 
3. We can't lose name associated with the order b/c they will be using that find orders. 

It seems like we are well past the time we should scratch this PR and reapproach with a clear solution.

**Comment by rymorgan** [comment]
The last column is still not fixed in place nor are the rows scrollable. 

qdgqxy6jwd

**Pull request closed without merging**